ecmascript/visit/macros/src/lib.rs

```diff
@@ -74,6 +75,46 @@ pub fn define(tts: proc_macro::TokenStream) -> proc_macro::TokenStream {
             path: q!({ allow }).parse(),
             tokens: q!({ (unused_variables) }).parse(),
         });
+
+        let fn_name = v.sig.ident.clone();
+        let default_body = replace(
+            &mut v.default,
+            Some(
+                q!(Vars { fn_name: &fn_name }, {
+                    {
+                        fn_name(self, n, _parent)
+                    }
+                })
+                .parse(),
+            ),
+        )
+        .clone();
+
+        let arg_ty = v
+            .sig
+            .inputs
+            .iter()
+            .skip(1)
+            .next()
+            .map(|v| match *v {
+                FnArg::Typed(ref pat) => &pat.ty,
+                _ => unreachable!(),
+            })
+            .unwrap();
+
+        tokens.push_tokens(&q!(
+            Vars {
+                fn_name,
+                default_body,
+                Type: arg_ty,
+            },
+            {
+                #[allow(unused_variables)]
+                fn fn_name<V: ?Sized + Visit>(_visitor: &mut V, n: Type, _parent: &dyn Node) {
```

Comment:
```suggestion
                pub fn fn_name<V: ?Sized + Visit>(_visitor: &mut V, n: Type, _parent: &dyn Node) {
```
With this change it's working as expected 👍

Reply:
Thanks for the review. I forgot exporting it.